frontend/src/models/system.js

```diff
@@ -187,6 +187,40 @@ const System = {
         return { success: false, error: e.message };
       });
   },
+  getWelcomeMessages: async function () {
+    return await fetch(`${API_BASE}/system/welcome-messages`, {
+      method: "GET",
+      cache: "no-cache",
+    })
+      .then((res) => {
+        if (!res.ok) throw new Error("Could not fetch welcome messages.");
+        return res.json();
+      })
+      .then((res) => res.welcomeMessages)
+      .catch((e) => {
+        console.error(e);
+        return null;
+      });
+  },
+  setWelcomeMessages: async function (messages) {
+    return fetch(`${API_BASE}/system/set-welcome-messages`, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({ messages }),
```

Comment:
setting welcome message, even in single user mode should add base headers because without that - even on a password protected instance anyone could update the messages if they knew the endpoint